### PR TITLE
Fix Serato Metadata checkbox behaviour

### PR DESCRIPTION
**Pull request by tech-bash**
Fix: #11226

Its working perfectly as wanted.

gsoc2

### CODE REVIEW

**Comment by uklotzde** [comment]
Closing PRs with unresolved review comments and opening new PRs instead is unhelpful.

**Comment by daschuer** [comment]
You need to start from the 2.4 branch to avoid rebasing.

**Comment by tech-bash** [comment]
> Closing PRs with unresolved review comments and opening new PRs instead is not helpful.

I know but sometimes it becomes quite messy, so I think of it as a fresh start. Sorry, will keep that mind for the future PRs.

**Comment by Aryan-Mishra24** [comment]
Man it's so hard to make head or tails of what's going on here

**Comment by daschuer** [comment]
You did your work however on the main branch. This will do the trick: 
`git rebase onto=upstream/2.3 HEAD~1`
`git push -f `

**Comment by Aryan-Mishra24** [comment]
Just so we're clear  are any additional steps or changes required before this patch can be merged.

File: src/preferences/dialog/dlgpreflibrary.cpp [hunk 1]
```diff
@@ -554,4 +555,6 @@ void DlgPrefLibrary::slotSyncTrackMetadataToggled() {
     if (isVisible() && checkBox_SyncTrackMetadata->isChecked()) {
         mixxx::DlgTrackMetadataExport::showMessageBoxOncePerSession();
     }
+    // If Track Metatdata checkbox is unchecked, ignore the state of Serato Metatdata checkbox.
```
Comment (uklotzde) [inline src/preferences/dialog/dlgpreflibrary.cpp:558]: It doesn't make sense to repeat and rephrase what the code does as a comment. And it is also inconsistent to insert a comment only here but not at the other place. Please remove it.

**Comment by uklotzde** [comment]
Please get familiar with Git before opening any new PRs. Such a trivial fix is not worth all this noise and waste of time.

**Comment by Aryan-Mishra24** [comment]
hit the hammer on the nail there bud i regret my unfamiliarity causing issues I'm running around trying to be helpful where i can but I've learnt more these 2 days then i learn't in months . And soon enough i'll be coding with 8 fingers instead of two right thumbs

**Comment by tech-bash** [comment]
> Please get familiar with Git before opening any new PRs. Such a trivial fix is not worth all this noise and waste of time for everyone that is involved.

Its not that I am not familiar with Git, its that I am trying to adapt Mixxx's coding style i know that in a few days I 'll be better, sorry for inconvenience!

**Comment by daschuer** [comment]
Do you have git installed locally? On which OS are you? 
Did you understand the commands from my post above? 
https://github.com/mixxxdj/mixxx/pull/11441#issuecomment-1496152280

If you feel uncertain with git. I strongly recommend to do:
http://git-school.github.io/visualizing-git/
It is a nice game and you will learn git on the run.

**Comment by tech-bash** [comment]
I am on Linux(Kubuntu), and I am travelling right now will push the changes after few hours, as I told ealier I am familiar with git, I understood the above command after the rebase there will be merge conflicts too that I will be needing to resolve and then I will push the updated changes, I am taking time because of my travelling issues.

**Comment by tech-bash** [comment]
Thank you for your concers!

**Comment by ronso0** [comment]
Thank you. Please squash the two commits to get rid of the kdev4 file.

**Comment by tech-bash** [comment]
What worked for me was: `git rebase --onto upstream/2.3 HEAD~1` and then `git push -f`, Thank you!

**Comment by ronso0** [comment]
Okay. Squashing can be done when rebasing **interactively** (-i)
`git rebase -i --onto upstream/2.3`
then just follow the instructions.

**Comment by daschuer** [comment]
@tech-bash 
Before merge, we need your permission to distribute the changes with Mixxx. 
Please sign https://docs.google.com/a/mixxx.org/spreadsheet/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ
and comment here when done.

**Comment by daschuer** [comment]
This is a last minute 2.3.5 candidate

**Comment by daschuer** [comment]
I con confirm that @tech-bash has signed the contributor agreement. 

Do you have interest to fix the CI so that we can include this in the next release?

**Comment by daschuer** [comment]
Closed via https://github.com/mixxxdj/mixxx/pull/11782. Thank you.

**Pull request closed without merging**